src/SFA.DAS.Commitments.Api.IntegrationTests/Tests/WhenSimulatingRealWorldApprenticeshipLoad.cs

```diff
@@ -0,0 +1,198 @@
+                .Concat(employerIdsPerTask.Select(ids => Task.Run(() => RepeatCallGetApprenticeships(ids))));
+
+            var callTimesPerTask = await Task.WhenAll(tasks);
+
+            var getApprenticeshipCalls = callTimesPerTask.Take(numberOfGetApprenticeshipTasks).SelectMany(cd => cd);
+            var slowestGetApprenticeshipCall = getApprenticeshipCalls.Max(d => d.CallTime);
+            var getApprenticeshipsCalls = callTimesPerTask.Skip(numberOfGetApprenticeshipTasks).SelectMany(cd => cd);
+            var slowestGetApprenticeshipsCall = getApprenticeshipsCalls.Max(d => d.CallTime);
+
+            var allCalls = getApprenticeshipCalls.Concat(getApprenticeshipsCalls);
+            await TestLog.Progress("Call Log:");
+            // if LogProgress is awaited, the details are written out-of-order
+            allCalls.OrderBy(c => c.StartTime).ForEach(c => TestLog.Progress(c.ToString()).GetAwaiter().GetResult());
+
+            TestLog.Progress($"Slowest GetApprenticeship  {slowestGetApprenticeshipCall}").GetAwaiter().GetResult();
+            TestLog.Progress($"Slowest GetApprenticeships {slowestGetApprenticeshipsCall}").GetAwaiter().GetResult();
+
+            Assert.LessOrEqual(slowestGetApprenticeshipCall, new TimeSpan(0, 0, 1));
+            Assert.LessOrEqual(slowestGetApprenticeshipsCall, new TimeSpan(0, 0, 1));
+        }
+
+        private class ApprenticeshipCallParams
+        {
+            public long ApprenticeshipId { get; set; }
+            public long EmployerId { get; set; }
+        }
+
+        private static async Task<CallDetails[]> RepeatCallGetApprenticeship(IEnumerable<ApprenticeshipCallParams> ids)
+        {
+            var tasks = ids.Select(i => CommitmentsApi.CallGetApprenticeship(i.ApprenticeshipId, i.EmployerId));
+            return await Task.WhenAll(tasks);
+        }
+
+        public static async Task<CallDetails[]> RepeatCallGetApprenticeships(IEnumerable<long> employerAccountIds)
+        {
+            //ideally want to use some sort of synchronization, so can kick this off in middle of getapprenticeship calls
+
+            //q&d
+            //Thread.Sleep(2 * 1000);
+
+            var tasks = employerAccountIds.Select(id => CommitmentsApi.CallGetApprenticeships(id));
+            return await Task.WhenAll(tasks);
+        }
+
+        /// <remarks>
+        /// It's better to fetch what we need from the generated data from the db, rather than...
+        /// 1) read *all* the db data into memory (a la provider events api in test), as would use an unnecessarily large amount of memory and be slow to read
+        /// 2) store it in the database in a similar way to the test ids, as that would require more code, increase db complexity etc. when you'd have to fetch the data from the db anyway
+        /// </remarks>
+        private async Task<IEnumerable<IEnumerable<ApprenticeshipCallParams>>> GetGetApprenticeshipDistinctCallParamsPerTask(int numberOfTasks, int getApprenticeshipCallsPerTask)
```

Comment:
this method isn't used. Can it be deleted?

Reply:
Yes it can, deleted.